docker/mongodb-kubernetes-tests/kubetester/mongodb_role.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional
+
+from kubeobject import CustomObject
+from kubetester.mongodb import MongoDBCommon, Phase, in_desired_state
+
+ClusterMongoDBRoleKind = "ClusterMongoDBRole"
+
+
+class ClusterMongoDBRole(CustomObject, MongoDBCommon):
+    def __init__(self, *args, **kwargs):
+        with_defaults = {
+            "plural": "clustermongodbroles",
+            "kind": "ClusterMongoDBRole",
+            "group": "mongodb.com",
+            "version": "v1",
+        }
+        with_defaults.update(kwargs)
+        super(ClusterMongoDBRole, self).__init__(*args, **with_defaults)
+
+    def assert_reaches_phase(self, phase: Phase, msg_regexp=None, timeout=None, ignore_errors=False):
```

Comment:
nit: is it worth adding a trace here ? or logging the reconciliation time
As we do e.g here: https://github.com/mongodb/mongodb-kubernetes/blob/6a1393da1c85ce7f02cc31986bc547bfd7adf03f/docker/mongodb-kubernetes-tests/kubetester/mongodb.py#L144